python/tests/test_tables.py

```diff
@@ -2336,6 +2344,34 @@ def test_copy(self):
         t1.edges.clear()
         assert t1 != t2
 
+    def test_equivalent(self):
+        pop_configs = [msprime.PopulationConfiguration(5) for _ in range(2)]
+        migration_matrix = [[0, 1], [1, 0]]
+        t1 = msprime.simulate(
+            population_configurations=pop_configs,
+            migration_matrix=migration_matrix,
+            mutation_rate=1,
+            record_migrations=True,
+            random_seed=1,
+        ).dump_tables()
+        t2 = msprime.simulate(
```

Comment:
why not just copy? this is also testing the ability of msprime to produce the same thing with the same random seed, which is beside the point here.